src/components/Switch.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+
+import Box from './Box';
+
+const HEIGHT = 16;
+const WIDTH = 27;
+const OFFSET = 1;
+const LENGTH = HEIGHT - OFFSET * 2;
+
+const Inner = styled(Box)`
+  transition: left 0.5s ease;
+`;
```

Comment:
I'd recommend decreasing the animation length to 200ms or 300ms.
500ms is quite slow for this type of animation and could make the UI feel less responsive to users.

Really good article on the subject here: https://uxdesign.cc/the-ultimate-guide-to-proper-use-of-animation-in-ux-10bd98614fa9